test/services/ProgressService/update.test.ts

```diff
@@ -0,0 +1,252 @@
+  beforeEach(async () => {
+    gatewayRepository = sinon.createStubInstance(GatewayRepository, {
+      get: Promise.resolve(gatewayInstance),
+    });
+
+    progressService = new ProgressService(
+      gatewayRepository,
+      originWeb3,
+      auxiliaryWeb3,
+      gatewayAddress,
+      originWorkerAddress,
+      auxiliaryWorkerAddress,
+    );
+
+    eip20GatewayMockObject = {
+      methods: {
+        progressStake: () => {},
+      },
+    };
+
+    progressStakeSpy = sinon.replace(
+      eip20GatewayMockObject.methods,
+      'progressStake',
+      sinon.fake.returns(progressStakeRawTx),
+    );
+
+    gatewaySpy = sinon.replace(
+      interacts,
+      'getEIP20Gateway',
+      sinon.fake.returns(eip20GatewayMockObject),
+    );
+
+    const eip20CoGatewayMockObject = {
+      methods: {
+        progressMint: () => {},
+      },
+    };
+    progressMintSpy = sinon.replace(
+      eip20CoGatewayMockObject.methods,
+      'progressMint',
+      sinon.fake.returns(progressMintRawTx),
+    );
+    coGatewaySpy = sinon.replace(
+      interacts,
+      'getEIP20CoGateway',
+      sinon.fake.returns(eip20CoGatewayMockObject as any),
+    );
+
+    utilsSpy = sinon.replace(
+      Utils,
+      'sendTransaction',
+      sinon.fake.returns(fakeTransactionHash),
+    );
+
+    message = sinon.createStubInstance(Message);
+    message.gatewayAddress = gatewayAddress;
+    message.messageHash = '0x4223A868';
+    message.hashLock = '0x123AAF';
+    message.isValidSecret.callsFake(() => true);
+  });
+
+  it('should progress on origin and auxiliary when source and target status '
+    + 'is declared', async () => {
+    message.sourceStatus = MessageStatus.Declared;
+    message.targetStatus = MessageStatus.Declared;
+
+    await progressService.update([message]);
+
+    SpyAssert.assert(
+      gatewayRepository.get,
+      1,
+      [[gatewayAddress]],
+    );
+
+    SpyAssert.assert(
+      utilsSpy,
+      2,
+      [
+        [progressStakeRawTx, { gasPrice: ORIGIN_GAS_PRICE, from: originWorkerAddress }],
+        [progressMintRawTx, { gasPrice: AUXILIARY_GAS_PRICE, from: auxiliaryWorkerAddress }],
+      ],
+    );
+
+    SpyAssert.assert(
+      progressMintSpy,
+      1,
+      [[message.messageHash, message.hashLock]],
+    );
+
+    SpyAssert.assert(
+      progressStakeSpy,
+      1,
+      [[message.messageHash, message.hashLock]],
+    );
+
+    SpyAssert.assert(
+      coGatewaySpy,
+      1,
+      [[auxiliaryWeb3, coGatewayAddress]],
+    );
+
+    SpyAssert.assert(
+      gatewaySpy,
+      1,
+      [[originWeb3, gatewayAddress]],
+    );
+
+    sinon.restore();
+  });
+
+  it('should progress on origin when source status is declared', async () => {
+    message.sourceStatus = MessageStatus.Declared;
+
+    await progressService.update([message]);
+
+    SpyAssert.assert(
+      utilsSpy,
+      1,
+      [[progressStakeRawTx, { gasPrice: ORIGIN_GAS_PRICE, from: originWorkerAddress }]],
+    );
+
+    SpyAssert.assert(
```

Comment:
Added assertion that cogateway and progressMint spy is not called.

Reply:
Done